### PR TITLE
[TIMOB-20440]: Remove unnecessary check on write method.

### PR DESCRIPTION
**Pull request by hieupham007**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-20440
Need to test https://github.com/appcelerator/titanium_mobile/pull/7671 as well, since the change was originally introduced there.

### CODE REVIEW

**Comment by ashcoding** [comment]
Could you also include tests in this folder as well? https://github.com/appcelerator/titanium_mobile/tree/master/ti_mocha_tests

Also with regards to this using the test case:-
```javascript
var win = Ti.UI.createWindow({
    backgroundColor : '#ffffff'
});
var button = Ti.UI.createButton({
    top : 10,
    width : 200,
    height : 50,
    title : "write file"
});
win.add(button);
button.addEventListener('click', function(e) {
    var f = Ti.Filesystem.getFile(Ti.Filesystem.applicationDataDirectory, 'fileter.txt');
    f.write('writing to the file would be enough to create it');
    alert(f.read());
});
 
win.open();
```

This is fine.

If I changed the directory to 
```javascript
var f = Ti.Filesystem.getFile(Ti.Filesystem.externalStorageDirectory, 'fileter.txt');
```
I will get the error (On Android 6.0 and above with no permission granted)
```
[ERROR] :  TiFileProxy: (KrollRuntimeThread) [8580,8580] IOException encountered
[ERROR] :  TiFileProxy: java.io.FileNotFoundException: /storage/emulated/0/sg.com.appcelerator/fileter.txt: open failed: EACCES (Permission denied)
```
Is this okay? Should the developer know that externalStorageDirectory requires permission in order to do this or should there be a way to handle this too?

**Comment by hieupham007** [comment]
Well that's expected. If they're opening a file that requires a permission, they would need permission. However, depending the location of that file, they may not need permissions to do it, hence this change. Since this is a Kroll method, we can't stop devs from calling it, so we would let them handle the logic.

**Comment by ashcoding** [comment]
Sounds good to me. :+1:
Code reviewed and functionally tested. Just needs tests in the ti_mocha_tests folder.

File: ti_mocha_tests/ti.filesystem.test.js [hunk 1]
```diff
@@ -431,6 +431,23 @@ describe("filesystem", function() {
 		}
 		finish();
 	});
+	//TIMOB-20440
+	it(“applicationDataDirectory”, function(finish) {
```
Comment (ashcoding) [inline ti_mocha_tests/ti.filesystem.test.js]: [ERROR] Unexpected character '“' [line 435, column 4]

I think you used the wrong double inverted commas here. Wrong character.

File: ti_mocha_tests/ti.filesystem.test.js [hunk 2]
```diff
@@ -431,6 +431,23 @@ describe("filesystem", function() {
...
+		var file = Ti.Filesystem.getFile(Ti.Filesystem.applicationDataDirectory, 'filter.txt');
+		var text = “testing a file”;
```
Comment (ashcoding) [inline ti_mocha_tests/ti.filesystem.test.js]: Same here. Wrong character.

**Comment by ashcoding** [comment]
`npm test` failing due to wrong character used.

File: ti_mocha_tests/ti.filesystem.test.js [hunk 3]
```diff
@@ -431,6 +431,23 @@ describe("filesystem", function() {
...
+		should(blob).not.be.null;
+		var readText = blob.text;
+		should(readText).not.be.null;
+		should(readText).not.be.type("undefined");
+		should(readText).be.a.String;
+		should(readText.length).eql(text.length);
+		should(readText).eql(text);
+		file.deleteFile();
+		finish();
+	}
```
Comment (ashcoding) [inline ti_mocha_tests/ti.filesystem.test.js]: Missing `);`

**Comment by ashcoding** [comment]
Code reviewed and functionally tested to be working.
**Accepted**
Test code has some typos. Will be doing a PR for test case after this.

**Comment by ashcoding** [comment]
Fix for Test case is here: https://github.com/appcelerator/titanium_mobile/pull/7871

**Pull request merged**